### PR TITLE
gptune: don't make `git` attribute an Executable

### PR DESCRIPTION
**Pull request by wdconinc**
In https://github.com/spack/spack/pull/45383, a `git = which("git")` was introduced in `gptune`, which overwrites the `git` attribute in the package. That has broken our packages.spack.io instance since `TypeError: Object of type Executable is not JSON serializable` (@alalazo, this is what I referred to in https://github.com/spack/packages.spack.io/pull/27).

This PR replaces the `git` command with a dict `cmd` with the commands as keys, and executables as values.

### CODE REVIEW

**Comment by wdconinc** [comment]
@spackbot fix style

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
> In #45383, a `git = which("git")` was introduced in `gptune`, which overwrites the `git` attribute in the package. That has broken our packages.spack.io instance since `TypeError: Object of type Executable is not JSON serializable` (@alalazo, this is what I referred to in [spack/packages.spack.io#27](https://github.com/spack/packages.spack.io/pull/27)).

🤦 Apologies...

File: var/spack/repos/builtin/packages/gptune/package.py
Comment (tgamblin) [inline var/spack/repos/builtin/packages/gptune/package.py:252]: For commands that are likely to come from outside spack, it is *far* better to prefer to do them in `python` (e.g. use (`shutil.rmtree` or `os.remove`) than to pull them from the shell environment. Can you use `os.remove` here?

File: var/spack/repos/builtin/packages/gptune/package.py
Comment (tgamblin) [inline var/spack/repos/builtin/packages/gptune/package.py:231]: can you use `shutil.copy` here?

File: var/spack/repos/builtin/packages/gptune/package.py
Comment (tgamblin) [inline var/spack/repos/builtin/packages/gptune/package.py:303]: This package should probably depend on `bash`. It's not guaranteed to be on the system.

File: var/spack/repos/builtin/packages/gptune/package.py
Comment (tgamblin) [inline var/spack/repos/builtin/packages/gptune/package.py:202]: Running `which` here means that you will get versions of these commands found when the package is imported, not versions that exist when the test environment set up (at which point test deps are activated).  In general I think it's bad practice to call `which()` outside the test environment itself.

**Review by tgamblin — COMMENTED** [review]
These are probably comments for the `gptune` maintainers, but I wonder if we can fix them here.

**Comment by tldahlgren** [comment]
@tgamblin Do you want to change your review from `comments` to `changes requested`?

**Comment by tgamblin** [comment]
No. I commented so that you could decide whether to fix this now or later.

**Comment by wdconinc** [comment]
I'm probably not the right person to make these changes, and I don't know if all `gptune` maintainers or contributors have write access to this branch under the spack organization.

**Comment by tgamblin** [comment]
The PR doesn't make it worse at least.  I'd say get it in but perhaps @liuyangzhuan could look at this for later.

**Comment by wdconinc** [comment]
Add @tgamblin's requests as issue at https://github.com/spack/spack/issues/46507 and will mark as resolved here.

**Pull request merged**
